sdk/data/azcosmos/ci.yml

```diff
@@ -70,5 +70,6 @@ extends:
               ServiceDirectory: 'data/azcosmos'
               Image: $(vm.image)
               GoVersion: '1.22.0'
+              EnableRaceDetector: true
```

Comment:
Doesn't look like this enabled the race detector. You can see the absence of `-race` in the `Run tests` stage. I _think_ this needs to be moved to the `parameters` block on line 25?

Reply:
It did, and actually failed the tests 

```
[Jun 13 21:53:06.734144] Retry: exit due to non-retriable status code
    testing.go:1398: race detected during execution of test
--- FAIL: TestItemConcurrent (0.63s)
```

I enabled it only on the Emulator run, because thats where the concurrent tests are running :)

Reply:
That made me discover that I had a bug on the test and I was concurrently writing on a shared variable

Reply:
Our ci.yml does 2 things, runs the normal tests through the arch template but that ignores the Emulator tests (can only run on Windows), then as a second stage, we run the other tests on Windows after installing the Emulator and here is where I added the flag because we want to catch probably the failures on real-ish scenarios.

Reply:
Reference to the build with `-race`: https://dev.azure.com/azure-sdk/public/_build/results?buildId=3872094&view=logs&j=07a8bf49-95b8-5c7c-de1a-77230f20b6cf&t=79c67422-3cb3-50fc-4ea3-00a141e27961&l=13

Reply:
Ah ok got it, thanks for clarifying.